airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryDestination.java

```diff
@@ -239,12 +240,34 @@ public AirbyteMessageConsumer getConsumer(JsonNode config,
       }
       final WriteDisposition syncMode = getWriteDisposition(configStream.getDestinationSyncMode());
 
-      writeConfigs.put(AirbyteStreamNameNamespacePair.fromAirbyteSteam(stream),
-          new BigQueryWriteConfig(TableId.of(schemaName, tableName), TableId.of(schemaName, tmpTableName), writer, syncMode, schema));
+      if (isGcsUploadingMode) {
+        GcsDestinationConfig gcsDestinationConfig = GcsDestinationConfig
+            .getGcsDestinationConfig(BigQueryUtils.getGcsJsonNodeConfig(config));
+        GcsCsvWriter gcsCsvWriter = initGcsWriter(gcsDestinationConfig, configStream);
+        gcsCsvWriter.initialize();
```

Comment:
In the initialize method we are using AmazonS3 client and I am not sure I follow why is that

Reply:
Hi @subodh1810. The answer here is similar to the previous comment. I partially re-used the already existing destination-gcs module (GCS destination CSV writer in particular). But destination-gcs had been implemented re-using already existed modules from destination-S3 as far as I understood. The reason is that the Amazon S3 client in most of the cases works for both GCS and S3 storages. It's better to ask @tuliren for details. 
So the idea is the next:
1. Re-using the existing destination-gcs (CSV writer) we upload data to GCS bucket
2. Using the native Bigquery writer we create a native job to migrate CSV formatted data from bucket to BigQuery